### ISSUE TITLE
Is it possible to inject into NSWindowController instances?

### ISSUE THREAD

**Issue by connorwyatt** [AliSoftware/Dip-UI#42]
Hi, I've been using Dip for a week or two with the UI extensions, and it's working great for my `NSViewController`s, but I haven't been able to find a way to inject into `NSWindowController`s so far.

It conforms to the `StoryboardInstantiatable` protocol, and there is a `dipTag` on the NSWindowController subclass but it's getting injected into. I also tried using the auto inject class to resolve the dependencies, and that was to no avail also.

I have a workaround in mind, by setting the isInitialController to false and creating the window in my app delegate in order to inject but it's not ideal.

I'm happy to concede that I have made a mistake somewhere, but I can't see it if I have.

Thanks in advance for any help you may be able to provide.

Connor

Edit: I can auto inject using the `Injected` class, it was a badly defined dependency, but would like to define the dependencies in the container initialisation.

Further edit: My above edit was incorrect, I still cannot get it to work.

P.S. I love how easy it was to implement, kudos on the framework, and I love the `weakSingleton` component scope, it makes sharing my view models really easy.

**Comment by ilyapuchka** [comment]
@connorwyatt I didn't have in mind support for NSWindowController, but if it's behaviour when instantiating from storyboard is the same you can implement support for it in extension the same way as for NSViewController

**Comment by connorwyatt** [comment]
Okay, no problem, I just wanted to check I wasn't missing anything, thanks.

**Comment by connorwyatt** [comment]
Update: I've rewritten my NSWindowController and recreated the storyboard after finding that the window controllers do get instantiated the same way as view controllers (with the user defined attributes being added to the NSObject) and it is now working. I have no idea what the problem was but it made me look deeper into the implementation and it's pretty impressive.

TLDR: You can inject into NSWindowControllers in the same way
